### PR TITLE
Complete most of documentation for makecpt

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Finish up aliasing **most** arguments for `makecpt`, improving the documentattion to be better than what we currently have at https://www.pygmt.org/v0.2.0/api/generated/pygmt.makecpt.html! For reference, upstream GMT docs is at https://docs.generic-mapping-tools.org/6.1/makecpt.html and https://github.com/GenericMappingTools/gmt/blame/master/doc/rst/source/makecpt.rst.

Live documentation preview is at https://pygmt-git-complete-docs-makecpt.gmt.vercel.app/api/generated/pygmt.makecpt.html

TODO:

Arguments/Aliases to document (see also https://www.generic-mapping-tools.org/GMT.jl/latest/makecpt/). New aliases are highlighted in **bold**:

- [x] A - **transparency**
- [x] C - cmap
- [x] D - **background**/~~**foreground**~~?
- [ ] [E](https://docs.generic-mapping-tools.org/6.1/makecpt.html#e) - **data_levels/nlevels**?
- [x] F - **color_model**
- [x] G - truncate
- [x] H - output
- [x] I - reverse
- [x] M - **overrule_bg**?
- [x] N - **no_bg**?
- [x] Q - **log**
- [ ] [S](https://docs.generic-mapping-tools.org/6.1/makecpt.html#s) - **auto**
- [x] T - series
- [x] V - verbose
- [x] W - **categorical**
- [x] Ww - **cyclic**
- [x] Z - continuous

Fixes #

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Comment by weiji14** [comment]
> * [ ]  [E](https://docs.generic-mapping-tools.org/6.1/makecpt.html#e) - **data_levels/nlevels**?
> * [ ]  [S](https://docs.generic-mapping-tools.org/6.1/makecpt.html#s) - **auto**

These two flags ([`-E`](https://docs.generic-mapping-tools.org/6.1/makecpt.html#e) and [`-S`](https://docs.generic-mapping-tools.org/6.1/makecpt.html#s)) require a **file** input (or **pandas.DataFrame** table), but this isn't implemented in `pygmt.makecpt` yet. Probably best to leave the implementation to a separate PR?

**Comment by seisman** [comment]
For `makecpt`, each option does too many things. Not sure if these are good choices:

``` 
D - background/foreground
W - categorical
```

**Comment by weiji14** [comment]
> For `makecpt`, each option does too many things. Not sure if these are good choices: 
> ``` 
> D - background/foreground
> W - categorical
> ```
Yeah, it does get confusing. I couldn't think of a word that covers background/foreground (and NaN) though. Same with W which could be categorical/wrap/cyclic :/

**Comment by seisman** [comment]
For **-W**, we could use `categorical` (True or False) for **-W** and `cyclic` (True or False) for **-Ww**.

**Comment by weiji14** [comment]
> For **-W**, we could use `categorical` (True or False) for **-W** and `cyclic` (True or False) for **-Ww**.

Sure, I did think of that actually, but we'll need an extra check so that categorical (W) and cyclic (Ww) are not used at the same time.

File: pygmt/mathops.py [hunk 1]
```diff
@@ -8,40 +8,106 @@
...
 @kwargs_to_strings(T="sequence", G="sequence")
 def makecpt(**kwargs):
     """
-    Creates a static color palette table (CPT).
+    Make GMT color palette tables.
+
+    This is a module that will help you make static color palette tables
+    (CPTs). In classic mode we write the CPT to standard output, while under
+    modern mode we simply save the CPT as the current session CPT (but see
+    *output*). You define an equidistant set of contour intervals or pass your
```
Comment (seisman) [inline pygmt/mathops.py:35]: Perhaps we should not mention *classic mode* here.

**Review by seisman — APPROVED** [review]

**Pull request merged**